nni/contrib/compression/quantization/dorefa_quantizer.py

```diff
@@ -54,39 +70,129 @@ def __init__(self, model: torch.nn.Module, config_list: List[Dict], evaluator: E
         super().__init__(model, config_list, evaluator, existed_wrappers=existed_wrappers)
         self.evaluator: Evaluator
         self.is_init = False
-
+        self.check_validation()
         self.register_dorefa_apply_method()
         self.register_track_func()
 
+    def check_validation(self):
+        for _, ts in self._target_spaces.items():
+            for _, target_space in ts.items():
+                assert target_space.quant_scheme != None
+                if target_space.type is TargetType.PARAMETER and target_space.quant_scheme != 'affine':
+                    warn_msg = f"Only supports affine mode for weight quantization, bug got {target_space.quant_scheme}"
+                    _logger.warning(warn_msg)
+                elif target_space.type is TargetType.OUTPUT:
+                    module = target_space._wrapper.module
+                    # case 1: activation module
+                    # case 2: module with activation fused_modules
+                    fused_modules = target_space._wrapper.fused_modules if target_space._wrapper.fused_modules else []
+                    if not isinstance(module, tuple(ACTIVATION_LIST)) and not (len(fused_modules) > 0 and # type: ignore
```

Comment:
Python do not recommend `len(list) > 0` ([pep8](https://peps.python.org/pep-0008/))